src/JustSaying/Messaging/MessageSerialization/NewtonsoftSerializer.cs

```diff
@@ -57,33 +57,30 @@ public MessageAttributes GetMessageAttributes(string message)
             return new MessageAttributes();
         }
 
-        var dict = new Dictionary<string, MessageAttributeValue>();
+        var attributes = new Dictionary<string, MessageAttributeValue>();
```

Comment:
Is it worth initialising the size of the dictionary with `props.Count()`?

Reply:
Is that "free", or does it have to enumerate the properties twice to count them? If the latter it might outweigh the benefit.

Reply:
I think something like this should work:

```
var attrObj = JObject.Parse(message).Value<JObject>("MessageAttributes");

if (attrObj.Count == 0)
{
   return new MessageAttributes(); 
}

var attributes = new Dictionary<string, MessageAttributeValue>(attrObj.Count);

foreach (var property in attrObj.Properties())...
```

Reply:
Ah, I didn't realise there was a property. I thought you meant LINQ count.

Reply:
I suggest doing that in the next PR which moves this code around anyway, then I don't need to worry about merge conflicts.

Reply:
It's just an `IEnumerble<T>`, so there's no `.Count` property, only the `.Count()` method, so it would need to be enumerated twice.